src/scripts/main.js

```diff
@@ -1,3 +1,230 @@
+
+      tile.id = r.toString() + '-' + c.toString();
+
+      const num = board[r][c];
+
+      updateTile(tile, num);
+      document.getElementById('board').append(tile);
+    }
+  }
+}
+
+function hasEmptyTile() {
+  for (let r = 0; r < rows; r++) {
+    for (let c = 0; c < columns; c++) {
+      if (board[r][c] === 0) {
+        return true;
+      }
+    }
+  }
+
+  return false;
+}
+
+function setTwo() {
+  if (!hasEmptyTile()) {
+    document.querySelector('.message-lose').classList.remove('hidden');
+
+    return;
+  }
+
+  let found = false;
+
+  while (!found) {
+    // random r, c
+    const r = Math.floor(Math.random() * rows); // 0-1 * 4 -> 0 - 4
+    const c = Math.floor(Math.random() * columns);
+
+    if (board[r][c] === 0) {
+      const randomNums = [2, 2, 2, 2, 2, 2, 2, 2, 2, 4];
+
+      board[r][c] = randomNums[Math.floor(Math.random() * 10)];
+
+      const tile = document.getElementById(r.toString() + '-' + c.toString());
+
+      tile.innerText = board[r][c];
+      tile.classList.add('x' + board[r][c].toString());
+      found = true;
+    }
+  }
+}
+
+function updateTile(tile, num) {
+  tile.innerText = '';
+  tile.classList.value = ''; // clear the classList
+  tile.classList.add('tile');
+
+  if (num > 0) {
+    tile.innerText = num;
+    tile.classList.add('x' + num.toString());
+  }
+}
+
+document.addEventListener('keyup', (e) => {
+  if (e.code === 'ArrowLeft' && button.classList.contains('restart')) {
+    slideLeft();
+    setTwo();
+  } else if (e.code === 'ArrowRight' && button.classList.contains('restart')) {
+    slideRight();
+    setTwo();
+  } else if (e.code === 'ArrowUp' && button.classList.contains('restart')) {
+    slideUp();
+    setTwo();
+  } else if (e.code === 'ArrowDown' && button.classList.contains('restart')) {
+    slideDown();
+    setTwo();
+  }
+
+  document.querySelector('.score').innerText = score;
+});
+
+function filterZero(argRow) {
+  return argRow.filter(num => num !== 0); // create a new array without zeroes
+}
+
+function slide(argRow) {
+  // [0, 2, 2, 2]
+  let copyRow = filterZero(argRow); // get rid of zeroes -> [2, 2, 2]
+
+  // slide
+  for (let i = 0; i < copyRow.length - 1; i++) {
+    // check every 2
+    if (copyRow[i] === copyRow[i + 1]) {
+      copyRow[i] *= 2;
+      copyRow[i + 1] = 0;
+      score += copyRow[i];
+    } // [2, 2, 2] -> [4, 0, 2]
+  }
+
+  copyRow = filterZero(copyRow); // [4, 2]
+
+  // add zeroes
+  while (copyRow.length < columns) {
+    copyRow.push(0);
+  } // [4, 2, 0, 0]
+
+  return copyRow;
+}
+
+function slideLeft() {
+  for (let r = 0; r < rows; r++) {
+    let row = board[r];
+
+    row = slide(row);
+    board[r] = row;
+
+    for (let c = 0; c < columns; c++) {
+      if (board[r][c] === 2048) {
+        document.querySelector('.message-win').classList.toggle('hidden');
+      }
+
+      const tile = document.getElementById(r.toString() + '-' + c.toString());
+      const num = board[r][c];
+
+      updateTile(tile, num);
+    }
```

Comment:
isn't it dupplicated code? check methods bellow⬇️ I suggest to move it into function and reuse, hm?)